content/en/BestPractices/prevent-security-misconfig.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Prevent Security Misconfiguration
+linkTitle: Prevent Security Misconfiguration
+weight: 630
+description: >
+  Configure your software properly to avoid security issues.
+---
+
+{{% pageinfo %}}
+In this article, we describe common security misconfigurations.
+{{% /pageinfo %}}
+
+OWASP includes security misconfiguration in their [Top 10 Web Application Security Risks (2021)](https://owasp.org/www-project-top-ten/).
+
+## Description
+
+Security misconfigurations are failures to implement the required security controls in a software component. Misconfigurations may lead to:
+
+- Disclosure of sensitive information
+- Data or system compromise
+
+## Examples
+
+Let's explore some common examples of software misconfiguration.
+
+### Debug Mode Enabled
+
+A developer may leave the debug mode enabled in the production environment. Debug information may contain sensitive data such as:
+
+- Application internal paths
+- Secret keys
+- Code snippets
+
+To prevent this:
+
+- Configure security settings for your language and framework
+- Disable the debug mode in the production environment
+     - For a .NET application, update the `Web.config` or `Machine.config` file:
+
+          ```
+          <compilation debug="false"/>
+          ```
+
+### Verbose Error Messages
+
+Similar to the [previous example](#debug-mode-enabled), an attacker may retrieve sensitive data from verbose error messages or a stack trace.
+
+To prevent this:
+
+- Don't expose a stack trace to users
+- Configure custom error messages for your software
+     - For a .NET application, modify the `Web.config` file:
+
+          ```
+          <system.web>
+          <customErrors mode="RemoteOnly" defaultRedirect="Error.aspx">
+          <error statusCode="401" redirect="CustomPage.aspx" />
+          <error statusCode="403" redirect="CustomPage.aspx" />
+          <error statusCode="404" redirect="NotFound.aspx" />
+          <error statusCode="500" redirect="Error.aspx" />
+          </customErrors>
+          </system.web>
+          ```
+
+### Directory Listing
+
+Directory listing allows users to see all files and directories stored on a web server. A malicious actor may get unauthorized access to sensitive files. Search the web with `intitle: "index of /"` to see websites that have directory listing enabled.
+
+In the following example, we show how to prevent this problem for an Apache Tomcat web server:
```

Comment:
I'm a little worried about this -- 

The "Apache Tomcat web server" and "Apache web server" are two different services. Maybe the simplest solution is:

```suggestion
In the following example, we show how to prevent this problem for a Tomcat web server:
```